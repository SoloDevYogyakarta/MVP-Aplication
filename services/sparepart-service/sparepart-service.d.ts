import { HttpStatus } from '@nestjs/common';
import { FreeTextEntity } from '../../database/entities/public/free-text-entity/free-text-entity';
import { SparepartField } from '../../dto/sparepart-dto/sparepart-dto';
export declare class SparepartService {
    private readonly logger;
    create(body: SparepartField[]): Promise<{
        status: HttpStatus;
        message: string;
        create: import("../../database/entities/services/sparepart-entity/sparepart-entity").SparepartInstance;
        free: FreeTextEntity;
    }>;
    update(id: number, body: SparepartField[]): Promise<{
        status: HttpStatus;
        message: string;
    }>;
    destroy(id: number): Promise<{
        status: HttpStatus;
        message: string;
    }>;
    freeDestroy(id: number): Promise<{
        status: HttpStatus;
        message: string;
    }>;
}
