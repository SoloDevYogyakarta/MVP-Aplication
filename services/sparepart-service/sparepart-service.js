"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var SparepartService_1;
Object.defineProperty(exports, "__esModule", { value: true });
exports.SparepartService = void 0;
const common_1 = require("@nestjs/common");
const free_text_entity_1 = require("../../database/entities/public/free-text-entity/free-text-entity");
const sparepart_entity_1 = require("../../database/entities/services/sparepart-entity/sparepart-entity");
let SparepartService = SparepartService_1 = class SparepartService {
    constructor() {
        this.logger = new common_1.Logger(SparepartService_1.name);
    }
    async create(body) {
        this.logger.log(SparepartService_1.name);
        let free;
        const create = await sparepart_entity_1.sparepartEntity.create();
        create.save();
        for (const values of body) {
            const freeText = await free_text_entity_1.freeTextEntity.create({
                ...values,
                sparepart_id: create.id,
            });
            freeText.save();
            free = freeText;
        }
        return {
            status: common_1.HttpStatus.CREATED,
            message: 'Sparepart has been create',
            create,
            free,
        };
    }
    async update(id, body) {
        this.logger.log(SparepartService_1.name);
        const findOne = await sparepart_entity_1.sparepartEntity.findOne({ where: { id } });
        if (!findOne) {
            throw new common_1.HttpException({
                status: common_1.HttpStatus.BAD_REQUEST,
                message: 'Sparepart not found',
            }, common_1.HttpStatus.BAD_REQUEST);
        }
        for (const values of body) {
            const free = await free_text_entity_1.freeTextEntity.findOne({ where: { id: values.id } });
            if (!free) {
                throw new common_1.HttpException({
                    status: common_1.HttpStatus.BAD_REQUEST,
                    message: 'Free text not found',
                }, common_1.HttpStatus.BAD_REQUEST);
            }
            free.text = values.text;
            free.save();
        }
        return { status: common_1.HttpStatus.OK, message: 'Sparepart has been update' };
    }
    async destroy(id) {
        this.logger.log(SparepartService_1.name);
        const findOne = await sparepart_entity_1.sparepartEntity.findOne({ where: { id } });
        if (!findOne) {
            throw new common_1.HttpException({
                status: common_1.HttpStatus.BAD_REQUEST,
                message: 'Sparepart not found',
            }, common_1.HttpStatus.BAD_REQUEST);
        }
        findOne.destroy();
        return { status: common_1.HttpStatus.OK, message: 'Sparepart has been delete' };
    }
    async freeDestroy(id) {
        const findOne = await free_text_entity_1.freeTextEntity.findOne({ where: { id } });
        if (!findOne) {
            throw new common_1.HttpException({
                status: common_1.HttpStatus.BAD_REQUEST,
                message: 'Free text not found',
            }, common_1.HttpStatus.BAD_REQUEST);
        }
        findOne.destroy();
        return { status: common_1.HttpStatus.OK, message: 'Free text has been delete' };
    }
};
exports.SparepartService = SparepartService;
exports.SparepartService = SparepartService = SparepartService_1 = __decorate([
    (0, common_1.Injectable)()
], SparepartService);
//# sourceMappingURL=sparepart-service.js.map