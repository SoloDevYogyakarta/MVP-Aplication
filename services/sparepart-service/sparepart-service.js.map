{"version":3,"file":"sparepart-service.js","sourceRoot":"","sources":["../../../src/services/sparepart-service/sparepart-service.ts"],"names":[],"mappings":";;;;;;;;;;AAAA,2CAA+E;AAC/E,uGAG0E;AAC1E,yGAAqG;AAI9F,IAAM,gBAAgB,wBAAtB,MAAM,gBAAgB;IAAtB;QACY,WAAM,GAAG,IAAI,eAAM,CAAC,kBAAgB,CAAC,IAAI,CAAC,CAAC;IAkF9D,CAAC;IAhFC,KAAK,CAAC,MAAM,CAAC,IAAsB;QACjC,IAAI,CAAC,MAAM,CAAC,GAAG,CAAC,kBAAgB,CAAC,IAAI,CAAC,CAAC;QACvC,IAAI,IAAqB,CAAC;QAC1B,MAAM,MAAM,GAAG,MAAM,kCAAe,CAAC,MAAM,EAAE,CAAC;QAC9C,MAAM,CAAC,IAAI,EAAE,CAAC;QACd,KAAK,MAAM,MAAM,IAAI,IAAI,EAAE;YACzB,MAAM,QAAQ,GAAG,MAAM,iCAAc,CAAC,MAAM,CAAC;gBAC3C,GAAG,MAAM;gBACT,YAAY,EAAE,MAAM,CAAC,EAAE;aACxB,CAAC,CAAC;YACH,QAAQ,CAAC,IAAI,EAAE,CAAC;YAChB,IAAI,GAAG,QAAQ,CAAC;SACjB;QACD,OAAO;YACL,MAAM,EAAE,mBAAU,CAAC,OAAO;YAC1B,OAAO,EAAE,2BAA2B;YACpC,MAAM;YACN,IAAI;SACL,CAAC;IACJ,CAAC;IAED,KAAK,CAAC,MAAM,CAAC,EAAU,EAAE,IAAsB;QAC7C,IAAI,CAAC,MAAM,CAAC,GAAG,CAAC,kBAAgB,CAAC,IAAI,CAAC,CAAC;QACvC,MAAM,OAAO,GAAG,MAAM,kCAAe,CAAC,OAAO,CAAC,EAAE,KAAK,EAAE,EAAE,EAAE,EAAE,EAAE,CAAC,CAAC;QACjE,IAAI,CAAC,OAAO,EAAE;YACZ,MAAM,IAAI,sBAAa,CACrB;gBACE,MAAM,EAAE,mBAAU,CAAC,WAAW;gBAC9B,OAAO,EAAE,qBAAqB;aAC/B,EACD,mBAAU,CAAC,WAAW,CACvB,CAAC;SACH;QACD,KAAK,MAAM,MAAM,IAAI,IAAI,EAAE;YACzB,MAAM,IAAI,GAAG,MAAM,iCAAc,CAAC,OAAO,CAAC,EAAE,KAAK,EAAE,EAAE,EAAE,EAAE,MAAM,CAAC,EAAE,EAAE,EAAE,CAAC,CAAC;YACxE,IAAI,CAAC,IAAI,EAAE;gBACT,MAAM,IAAI,sBAAa,CACrB;oBACE,MAAM,EAAE,mBAAU,CAAC,WAAW;oBAC9B,OAAO,EAAE,qBAAqB;iBAC/B,EACD,mBAAU,CAAC,WAAW,CACvB,CAAC;aACH;YACD,IAAI,CAAC,IAAI,GAAG,MAAM,CAAC,IAAI,CAAC;YACxB,IAAI,CAAC,IAAI,EAAE,CAAC;SACb;QACD,OAAO,EAAE,MAAM,EAAE,mBAAU,CAAC,EAAE,EAAE,OAAO,EAAE,2BAA2B,EAAE,CAAC;IACzE,CAAC;IAED,KAAK,CAAC,OAAO,CAAC,EAAU;QACtB,IAAI,CAAC,MAAM,CAAC,GAAG,CAAC,kBAAgB,CAAC,IAAI,CAAC,CAAC;QACvC,MAAM,OAAO,GAAG,MAAM,kCAAe,CAAC,OAAO,CAAC,EAAE,KAAK,EAAE,EAAE,EAAE,EAAE,EAAE,CAAC,CAAC;QACjE,IAAI,CAAC,OAAO,EAAE;YACZ,MAAM,IAAI,sBAAa,CACrB;gBACE,MAAM,EAAE,mBAAU,CAAC,WAAW;gBAC9B,OAAO,EAAE,qBAAqB;aAC/B,EACD,mBAAU,CAAC,WAAW,CACvB,CAAC;SACH;QACD,OAAO,CAAC,OAAO,EAAE,CAAC;QAClB,OAAO,EAAE,MAAM,EAAE,mBAAU,CAAC,EAAE,EAAE,OAAO,EAAE,2BAA2B,EAAE,CAAC;IACzE,CAAC;IAED,KAAK,CAAC,WAAW,CAAC,EAAU;QAC1B,MAAM,OAAO,GAAG,MAAM,iCAAc,CAAC,OAAO,CAAC,EAAE,KAAK,EAAE,EAAE,EAAE,EAAE,EAAE,CAAC,CAAC;QAChE,IAAI,CAAC,OAAO,EAAE;YACZ,MAAM,IAAI,sBAAa,CACrB;gBACE,MAAM,EAAE,mBAAU,CAAC,WAAW;gBAC9B,OAAO,EAAE,qBAAqB;aAC/B,EACD,mBAAU,CAAC,WAAW,CACvB,CAAC;SACH;QACD,OAAO,CAAC,OAAO,EAAE,CAAC;QAClB,OAAO,EAAE,MAAM,EAAE,mBAAU,CAAC,EAAE,EAAE,OAAO,EAAE,2BAA2B,EAAE,CAAC;IACzE,CAAC;CACF,CAAA;AAnFY,4CAAgB;2BAAhB,gBAAgB;IAD5B,IAAA,mBAAU,GAAE;GACA,gBAAgB,CAmF5B"}