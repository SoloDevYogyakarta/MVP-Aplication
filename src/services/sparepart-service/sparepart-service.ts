import { HttpException, HttpStatus, Injectable, Logger } from '@nestjs/common';
import {
  FreeTextEntity,
  freeTextEntity,
} from '../../database/entities/public/free-text-entity/free-text-entity';
import { sparepartEntity } from '../../database/entities/services/sparepart-entity/sparepart-entity';
import { SparepartField } from '../../dto/sparepart-dto/sparepart-dto';

@Injectable()
export class SparepartService {
  private readonly logger = new Logger(SparepartService.name);

  async create(body: SparepartField[]) {
    this.logger.log(SparepartService.name);
    let free!: FreeTextEntity;
    const create = await sparepartEntity.create();
    create.save();
    for (const values of body) {
      const freeText = await freeTextEntity.create({
        ...values,
        sparepart_id: create.id,
      });
      freeText.save();
      free = freeText;
    }
    return {
      status: HttpStatus.CREATED,
      message: 'Sparepart has been create',
      create,
      free,
    };
  }

  async update(id: number, body: SparepartField[]) {
    this.logger.log(SparepartService.name);
    const findOne = await sparepartEntity.findOne({ where: { id } });
    if (!findOne) {
      throw new HttpException(
        {
          status: HttpStatus.BAD_REQUEST,
          message: 'Sparepart not found',
        },
        HttpStatus.BAD_REQUEST,
      );
    }
    for (const values of body) {
      const free = await freeTextEntity.findOne({ where: { id: values.id } });
      if (!free) {
        throw new HttpException(
          {
            status: HttpStatus.BAD_REQUEST,
            message: 'Free text not found',
          },
          HttpStatus.BAD_REQUEST,
        );
      }
      free.text = values.text;
      free.save();
    }
    return { status: HttpStatus.OK, message: 'Sparepart has been update' };
  }

  async destroy(id: number) {
    this.logger.log(SparepartService.name);
    const findOne = await sparepartEntity.findOne({ where: { id } });
    if (!findOne) {
      throw new HttpException(
        {
          status: HttpStatus.BAD_REQUEST,
          message: 'Sparepart not found',
        },
        HttpStatus.BAD_REQUEST,
      );
    }
    findOne.destroy();
    return { status: HttpStatus.OK, message: 'Sparepart has been delete' };
  }

  async freeDestroy(id: number) {
    const findOne = await freeTextEntity.findOne({ where: { id } });
    if (!findOne) {
      throw new HttpException(
        {
          status: HttpStatus.BAD_REQUEST,
          message: 'Free text not found',
        },
        HttpStatus.BAD_REQUEST,
      );
    }
    findOne.destroy();
    return { status: HttpStatus.OK, message: 'Free text has been delete' };
  }
}
