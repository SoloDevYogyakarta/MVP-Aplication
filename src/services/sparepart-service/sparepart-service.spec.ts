import { SparepartService } from './sparepart-service';

describe('SparepartService', () => {
  let service: SparepartService;

  beforeEach(() => {
    service = new SparepartService();
  });

  it('should to be defined', () => expect(service).toBeDefined());

  it('render correctly', () => expect(service).toMatchSnapshot());
});
