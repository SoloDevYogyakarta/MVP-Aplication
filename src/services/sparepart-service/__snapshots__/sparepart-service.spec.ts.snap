// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SparepartService render correctly 1`] = `
SparepartService {
  "logger": Logger {
    "context": "SparepartService",
    "options": {},
  },
}
`;
